src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/TableSchemaObject.java

```diff
@@ -1,22 +1,107 @@
 
-  public TableSchemaObject(TableMetadata tableMetadata) {
+  private final VectorConfig vectorConfig;
+
+  private TableSchemaObject(TableMetadata tableMetadata, VectorConfig vectorConfig) {
     super(TYPE, tableMetadata);
+    this.vectorConfig = vectorConfig;
   }
 
   @Override
   public VectorConfig vectorConfig() {
-    return VectorConfig.notEnabledVectorConfig();
+    return vectorConfig;
   }
 
   @Override
   public IndexUsage newIndexUsage() {
     return IndexUsage.NO_OP;
   }
+
+  /**
+   * Get table schema object from table metadata
+   *
+   * @param tableMetadata
+   * @param objectMapper
+   * @return
+   */
+  public static TableSchemaObject getTableSettings(
+      TableMetadata tableMetadata, ObjectMapper objectMapper) {
+    Map<String, String> extensions =
+        (Map<String, String>)
+            tableMetadata.getOptions().get(CqlIdentifier.fromInternal("extensions"));
+    String vectorize = extensions != null ? extensions.get("vectorize") : null;
+    Map<String, VectorConfig.ColumnVectorDefinition.VectorizeConfig> resultMap = new HashMap<>();
+    if (vectorize != null) {
+      try {
+        String vectorizeJson =
+            new String(ByteUtils.fromHexString(vectorize).array(), StandardCharsets.UTF_8);
+        // Convert JSON string to Map
+        JsonNode vectorizeByColumns = objectMapper.readTree(vectorizeJson);
+        Map<String, VectorConfig.ColumnVectorDefinition.VectorizeConfig> vectorizeConfigMap =
+            new HashMap<>();
+        while (vectorizeByColumns.fields().hasNext()) {
+          Map.Entry<String, JsonNode> entry = vectorizeByColumns.fields().next();
+          VectorConfig.ColumnVectorDefinition.VectorizeConfig vectorizeConfig =
+              objectMapper.treeToValue(
+                  entry.getValue(), VectorConfig.ColumnVectorDefinition.VectorizeConfig.class);
+          vectorizeConfigMap.put(entry.getKey(), vectorizeConfig);
+        }
+      } catch (JsonProcessingException | IllegalArgumentException e) {
+        throw SchemaException.Code.INVALID_VECTORIZE_CONFIGURATION.get();
+      }
+    }
+    VectorConfig vectorConfig;
+    List<VectorConfig.ColumnVectorDefinition> columnVectorDefinitions = new ArrayList<>();
+    for (Map.Entry<CqlIdentifier, ColumnMetadata> column : tableMetadata.getColumns().entrySet()) {
+      if (column.getValue().getType() instanceof VectorType vectorType) {
+        final Optional<IndexMetadata> index = tableMetadata.getIndex(column.getKey());
+        SimilarityFunction similarityFunction = SimilarityFunction.COSINE;
+        if (index.isPresent()) {
+          final IndexMetadata indexMetadata = index.get();
+          final Map<String, String> indexOptions = indexMetadata.getOptions();
+          final String similarityFunctionValue = indexOptions.get("similarity_function");
+          if (similarityFunctionValue != null) {
+            similarityFunction = SimilarityFunction.fromString(similarityFunctionValue);
```

Comment:
Can this throw an exception for bad similarity function value? If so, should catch.

Reply:
Shouldn't have error, those are saved from the request which are already validated.

Reply:
Never say never :-)
But I guess it's unlikely enough not to matter until we actually hit that issue.